### PR TITLE
Testing and a bit of backend refactoring

### PR DESCRIPTION
**Pull request by matthewhung09**

### CODE REVIEW

File: backend/models/user.js [hunk 1]
```diff
@@ -35,6 +35,6 @@ UserSchema.pre("save", async function (next) {
   next();
 });
 
-UserSchema.plugin(uniqueValidator);
+// UserSchema.plugin(uniqueValidator);
```
Comment (gscotti6950) [inline backend/models/user.js:38]: we should ask the professor if we should keep this in

**Review by gscotti6950 — COMMENTED** [review]
Looks good. I was there when you were making most of these changes so there wasn't much I needed to comment on. Good work!

**Comment by naomidonato** [comment]
Great job refactoring tests in order to pass CI check.

**Review by naomidonato — APPROVED** [review]
Backend tests have been refactored conscientiously.

**Comment by adellevo** [comment]
Nice job! We talked about the improvements in class, so I don't have any feedback for changes.

**Review by adellevo — COMMENTED** [review]
Nice job! We talked about the improvements in class, so I don't have any feedback for changes.

**Pull request merged**